crates/red_knot_python_semantic/src/types.rs

```
@@ -1339,96 +1434,201 @@ impl<'db> Type<'db> {
+            Type::StringLiteral(_) | Type::LiteralString => {
+                KnownClass::Str.to_instance(db).static_member(db, name)
             }
 
-            Type::LiteralString => {
-                // TODO defer to `typing.LiteralString`/`builtins.str` methods
-                // from typeshed's stubs
-                Symbol::todo("Attribute access on `LiteralString` types")
-            }
-
-            Type::BytesLiteral(_) => KnownClass::Bytes.to_instance(db).member(db, name),
+            Type::BytesLiteral(_) => KnownClass::Bytes.to_instance(db).static_member(db, name),
 
             // We could plausibly special-case `start`, `step`, and `stop` here,
             // but it doesn't seem worth the complexity given the very narrow range of places
             // where we infer `SliceLiteral` types.
-            Type::SliceLiteral(_) => KnownClass::Slice.to_instance(db).member(db, name),
+            Type::SliceLiteral(_) => KnownClass::Slice.to_instance(db).static_member(db, name),
 
-            Type::Tuple(_) => {
-                // TODO: implement tuple methods
-                Symbol::todo("Attribute access on heterogeneous tuple types")
-            }
+            Type::Tuple(_) => KnownClass::Tuple.to_instance(db).static_member(db, name),
 
             Type::AlwaysTruthy | Type::AlwaysFalsy => match name {
                 "__bool__" => {
                     // TODO should be `Callable[[], Literal[True/False]]`
                     Symbol::todo("`__bool__` for `AlwaysTruthy`/`AlwaysFalsy` Type variants")
                 }
-                _ => Type::object(db).member(db, name),
+                _ => Type::object(db).static_member(db, name),
+            },
+        }
+    }
+
+    /// Call the `__get__(instance, owner)` method on a type, if it exists.
+    fn try_call_dunder_get(
+        &self,
+        db: &'db dyn Db,
+        instance: Option<Type<'db>>,
+        owner: Type<'db>,
+    ) -> Option<Type<'db>> {
```

I don't think this needs to be done but I've a slight preference if this method would already return a `Result` and
that we instead add a `TODO` at the call sites that suppress the error handling. Doing this now also ensures that new call sites *have* to think about handling the error or it requires an explicit TODO comment if they can't (e.g. because it's a call in `.memeber`)